clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2288,23 +2253,15 @@ public void testRebalanceException(GroupProtocol groupProtocol) {
         client.prepareResponseFrom(syncGroupResponse(singletonList(tp0), Errors.NONE), coordinator);
 
         // assign throws
-        try {
-            consumer.updateAssignmentMetadataIfNeeded(time.timer(Long.MAX_VALUE));
-            fail("Should throw exception");
-        } catch (Throwable e) {
-            assertEquals(partitionAssigned + singleTopicPartition, e.getCause().getMessage());
-        }
+        Throwable t = assertThrows(Throwable.class, () -> consumer.updateAssignmentMetadataIfNeeded(time.timer(Long.MAX_VALUE)));
+        assertEquals(partitionAssigned + singleTopicPartition, t.getCause().getMessage());
 
         // the assignment is still updated regardless of the exception
         assertEquals(singleton(tp0), subscription.assignedPartitions());
 
         // close's revoke throws
-        try {
-            consumer.close(Duration.ofMillis(0));
-            fail("Should throw exception");
-        } catch (Throwable e) {
-            assertEquals(partitionRevoked + singleTopicPartition, e.getCause().getCause().getMessage());
-        }
+        t = assertThrows(Throwable.class, () -> consumer.close(Duration.ofMillis(0)));
```

Comment:
Not introduced by this PR, but here we could be more specific and `assertThrows(KafkaException.class, ...)`. Regardless of the type of exception thrown in the rebalance callbacks, a `KafkaException` is what's always propagated (legacy and new consumer).

Btw, reviewing this PR I noticed that this KafkaConsumerTest has lots of tests not enabled for the new consumer (with lots of TODOs about it). I created [KAFKA-16737](https://issues.apache.org/jira/browse/KAFKA-16737) to make sure we address it, enable the tests that apply and clean up the code.

Reply:
Good call, I pushed an update.
Thanks for raising KAFKA-16737